test/system/700-play.bats

```diff
@@ -418,6 +418,38 @@ status: {}
     run_podman rm -f -t0 myyaml
 }
 
+@test "podman play with init container" {
+    TESTDIR=$PODMAN_TMPDIR/testdir
+    mkdir -p $TESTDIR
+
+testUserYaml="
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+spec:
+  containers:
+  - command:
+    - ls
+    - /dev/shm/test1
+    image: $IMAGE
+    name: testCtr
+  initContainers:
+  - command:
+    - touch
+    - /dev/shm/test1
+    image: $IMAGE
+    name: initCtr
+"
+    echo "$testUserYaml" > $PODMAN_TMPDIR/test.yaml
+    run_podman kube play $PODMAN_TMPDIR/test.yaml
+    assert "$output" !~ "ERRO" "init containers should not generate logrus.Error"
```

Comment:
I think output only includes stdout. @edsantiago do we have another variable for stderr?

Reply:
Unfortunately, no. BATS `$output` is always stdout + stderr combined, no way to separate them. So:
* if your question is, can we check stdout on its own, the answer is no
* if your question is, can we check for the presence of an error message, then yes.

Reply:
Thanks, Ed.  My suspicion was wrong then. Do you have an idea why this check does *not* fail with a previous podman?

Reply:
Changing `!~` to `=~` gives the answer:
```
  #| expected: =~ 'ERRO'
   #|   actual:    'time="2022-12-08T05:30:23-07:00" level=error msg="Removing container 9bfec1031db0fd30a9531104cea582cd17cd90f4127ffd8aec26d411fc252b43 from database: no container with ID 9bfec1031db0fd30a9531104cea582cd17cd90f4127ffd8aec26d411fc252b43 found in DB: no such container"'
```
One possible fix:
```diff
-    assert "$output" !~ "ERRO" "init containers should not generate logrus.Error"
+    assert "$output" !~ "level=" "init containers should not generate logrus.Error"
```

Reply:
PS @vrothberg **thank you** for checking against baseline.

Reply:
diff --git a/libpod/pod_api.go b/libpod/pod_api.go
index 7b143b478c0d..08a2e142948e 100644
--- a/libpod/pod_api.go
+++ b/libpod/pod_api.go
@@ -44,10 +44,6 @@ func (p *Pod) startInitContainers(ctx context.Context) error {
                                icLock.Unlock()
                                return fmt.Errorf("failed to remove once init container %s: %w", initCon.ID(), err)
                        }
-                       // Removing a container this way requires an explicit call to clean up the db
-                       if err := p.runtime.state.RemoveContainerFromPod(p, initCon); err != nil {
-                               logrus.Errorf("Removing container %s from database: %v", initCon.ID(), err)
-                       }
                        icLock.Unlock()
                }
        }

Doesn't this leaves us in a weird state where a container no longer exists but is present in a Pod? Once this initContainer is removed, it should no longer be listed in the pod?

Reply:
> Doesn't this leaves us in a weird state where a container no longer exists but is present in a Pod? Once this initContainer is removed, it should no longer be listed in the pod?

The `removeContainer` call above takes care of that.